### ISSUE TITLE
Create a responsive navigation sidebar

### ISSUE THREAD

**Issue by SunderB** [SunderB/sonic-pi-phoenix-website#4]
## Aim
Create a responsive navigation sidebar suitable for both desktop/laptop screens and mobile devices.
* On larger screens show a sidebar beside the main content - should be hidden on mobile
* On mobile devices show a menu button which opens a navigation menu

## Examples
* [Godot Engine](https://docs.godotengine.org/en/stable/index.html) has one large treeview sidebar listing all pages. On small screens the sidebar disappears and is revealed when the use clicks the menu button
* [Matplotlib](https://matplotlib.org/stable/api/index.html) has multiple seperate sections on the top nav bar, then has a sidebar with a treeview listing all the categories and pages in the section

## Ideas
* Use the [slideover component](https://petal.build/components/slide_over) in Petal Components on mobile devices
* Have all subcategories and pages listed in a treeview like layout
   * Possibly a variant of an 'Accordion' component acting as the tree-structured navigation where multiple sections can be expanded at once. This isn't currently available in the library. They are themselves working on an accordion, but allowing multiple sections to be expanded at once may or may not make it into the component. They have said they will look into allowing it as an option.
* Use CSS flexbox to split space between sidebar and main content on larger screens, and to show/hide the menu button(?)
   * Tailwind's screen size classes might be useful? https://tailwindcss.com/docs/responsive-design
* Have tabs for Tutorial, Reference and Examples(?)

Relates to #12.

**Comment by ethancrawford** [comment]
Do you have something specific in mind?

**Comment by SunderB** [comment]
Copied from #12 for easy reference:
@ethancrawford:
The current technology choice for this mobile-friendly refactor is to use [Petal components](http://petal.build/components).
Re the navigation menus, we may not need to have two separate menus. My proposed implementation is to have a vertical, tree-structured navigation menu. In mobile view, this would involve using something like a 'SlideOver' component, equivalent to Bootstrap's 'Offcanvas', to create a slide-in panel. This could then be combined with a variant of an 'Accordion' component acting as the tree-structured navigation where multiple sections can be expanded at once.

As of this moment, neither of these components are currently available in the Petal components library, so it will involve some custom work on our side. I have almost finished building an implementation of the SlideOver component, which Petal folks have said they are happy to receive. They are themselves working on an accordion, but allowing multiple sections to be expanded at once may or may not make it into the component. They have said they will look into allowing it as an option.

**Comment by ethancrawford** [comment]
Here are examples of the kind of design structure I have in mind: (exact content and style will of course)
![mobile nav](https://user-images.githubusercontent.com/10395940/165939405-d4475a7b-2c33-4f3b-aa3e-8e4255d17978.gif)

![desktop nav](https://user-images.githubusercontent.com/10395940/165939423-19993c4c-1a0a-40b6-b049-059049a58c6e.gif)

Would appreciate your thoughts also @samaaron
